neural_structured_learning/examples/notebooks/adversarial_cnn_transfer_learning_fashionmnist.ipynb

```diff
@@ -0,0 +1,1263 @@
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "W2KeGFC96_n5"
+      },
+      "source": [
+        "## Adversarial Attack Examples\n",
+        "\n",
+        "Here we look at a few examples of applying thte FGSM adversarial attack on sample apparel images and how it affects our model predictions. We create a simple wrapper over our `perform_adversarial_attack_fgsm` function to try it out on sample images."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "d-yy8_Sc7Ihy"
+      },
+      "source": [
+        "def show_adv_attack_example(image_idx, image_dataset, \n",
+        "                            image_labels, cnn_model,\n",
+        "                            class_names, loss_fn, eps):\n",
+        "  sample_apparel_img = image_dataset[sample_apparel_idx]\n",
+        "  sample_apparel_labelidx = image_labels[sample_apparel_idx]\n",
+        "  perform_adversarial_attack_fgsm(input_image=sample_apparel_img, \n",
+        "                                  image_label_idx=sample_apparel_labelidx, \n",
+        "                                  cnn_model=cnn_model, \n",
+        "                                  class_names_map=class_names,\n",
+        "                                  loss_func=loss_fn, eps=eps)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "nnvwU1KJvIcc"
+      },
+      "source": [
+        "show_adv_attack_example(6, test_images_3ch_scaled, \n",
+        "                        test_labels, model,\n",
+        "                        class_names, scc, 0.05)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "ubz5GhmjB_VG"
+      },
+      "source": [
+        "show_adv_attack_example(60, test_images_3ch_scaled, \n",
+        "                        test_labels, model,\n",
+        "                        class_names, scc, 0.05)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "273SaMunCUfF"
+      },
+      "source": [
+        "show_adv_attack_example(500, test_images_3ch_scaled, \n",
+        "                        test_labels, model,\n",
+        "                        class_names, scc, 0.05)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "K1eWaNaVXFCR"
+      },
+      "source": [
+        "show_adv_attack_example(560, test_images_3ch_scaled, \n",
+        "                        test_labels, model,\n",
+        "                        class_names, scc, 0.05)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "0ZojtMft7LRd"
+      },
+      "source": [
```

Comment:
The perturbed test dataset generation would better belong towards the end of the tutorial right before you evaluate the base and adversarial model with the perturbed test dataset.

Reply:
got it! moved this